Landing page donate

lehysen opened the pull request.

lehysen committed:
agregando boton de donate al html

lehysen committed:
agregando estilos al boton

misscs commented:
@lehysen Few things to remember

1. When you're PR is ready for review, request the reviewer so that they are notified
2. Always title case your PR commit messages
3. We use the present/imperative tense for PR commit messages, not the past tense. You want to answer the question _This PR...._ in your minds. Example, Agrega instead of Agregando

misscs commented on app/assets/styles/styles.css:
```diff
@@ -170,3 +170,26 @@ a {
+  text-decoration: none;
+  color: white;
+  background-color: #6fcf97;
+  padding: 15px 40px;
```
We define padding in `em` values

lehysen replied:
**Domain:** hesitant-ink.surge.sh

Ese es el dominio en surge.

misscs commented on index.html:
```diff
@@ -87,7 +87,13 @@ <h1 class="logo-brand" itemprop="brand">MACILE</h1>
   </nav>
-
+  <div class="donate">
+    <div>
```
Why do we need another `div` inside the parent?

misscs commented on app/assets/styles/styles.css:
```diff
@@ -170,3 +170,26 @@ a {
+  font-size: 1.5rem;
+  margin-left: 2em;
+  margin-right: 2em;
+  background-image: linear-gradient(rgba(21, 21, 78, 0.5), rgba(21, 21, 78, 0.6)), url("/home/salvia/news.macile.org/app/assets/images/salvia-team.webp");
```
The image path needs to be absolute from the project directory not your computer directory

misscs reviewed (changes requested):
A few small changes. 

Can you push your work to surge? You will need to install it on your machine following these directions here ... http://surge.sh/help/getting-started-with-surge Step #2. Your computer already has node installed. Just follow the command. 

Once you install surge you will need to create an account. Then you can publish your pages.

lehysen committed:
agrega cambios al index.html y al style.css

misscs commented on app/.vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "git.ignoreLimitWarning": true
+}
```
This file should not be in the repo. Make sure you don't use `git add .` I always like to double check what files changed. 

You may need to add this file to `.gitignore`. This is a file that that lives in the project root.

misscs commented on package.json:
```diff
@@ -24,5 +24,8 @@
     "imagemin-mozjpeg": "^8.0.0",
     "imagemin-pngquant": "^8.0.0",
     "imagemin-webp": "^5.1.0"
+  },
+  "dependencies": {
+    "http-server": "^0.12.3"
```
We should be careful not to add `http-server` as a project dependency. That means that it is required by the project in production. 

In general, I don't think we want to add it as a dev dependency either. It is just a tool that you need on your machine. 

You need to remove lines 28-30. Then run `npm install` so it removes the dependency from `package-lock.json`. Commit the changes to `package-lock.json`. 

You probably need to install the server globally `npm install http-sever -g`. That makes it global on your machine.

misscs commented:
Don't forget that your commit messages need to start with a capital letter!

misscs commented:
Thank you for pushing your work to surge. It looks pretty good. I would like to request a few changes so that that design really pops. 

## Add more space between the text and button

In the mockup, the Donate button is not flush with the text.
- Add `display: block` to the donate button. This ensures that the link behaves like a block level element. Without this, if you add margin to the `p` it won't visually apply.
- Add a class to the `p` element and then you can add margin bottom

misscs commented on app/assets/styles/styles.css:
```diff
@@ -170,3 +170,26 @@ a {
 }
+
+.donate {
+  height: 300px;
```
In general, it is best to avoid a fixed height on elements. I like to use padding because it creates a more fluid scaling experience. 300px may look good on desktop, but it takes up too much space on mobile. 

Let's remove this property and add `padding: 1em 3em; for large screens and `padding: 1em` for small screens.

misscs commented on app/assets/styles/styles.css:
```diff
@@ -170,3 +170,26 @@ a {
+
+.donate {
+  height: 300px;
+  font-size: 1.5rem;
```
It isn't recommended to apply a font-size to the parent element unless all the children use the same font-size or we are using ems/% based fonts and want to scale from a parent/root element. 

Since our code base is more explicit, we want to apply `font-size` directly on the elements we want to target, the title, subtitle, and button, in the case of the donate box.

misscs commented on app/assets/styles/styles.css:
```diff
@@ -170,3 +170,26 @@ a {
+  height: 300px;
+  font-size: 1.5rem;
+  margin-left: 2em;
+  margin-right: 2em;
```
We're going to need to define a max-width. I didn't really think about one when I making the designs. For now, because we know it will need to sit in the center let's replace `2em` with `auto`. 

Once we merge everything in, I'll take a cleanup pass to add max-widths to containers.

misscs commented on app/assets/styles/styles.css:
```diff
@@ -170,3 +170,26 @@ a {
+}
+
+.donate__btn {
+  font-size: 1.5rem;
```
This font size is a bit big. Let's remove it for now.

misscs commented on app/assets/styles/styles.css:
```diff
@@ -170,3 +170,26 @@ a {
+  text-decoration: none;
+  color: white;
+  background-color: #6fcf97;
+  padding: 1em 1.5em;
```
There's a bit too much padding on the button. Try `0.5em 1em`

misscs commented on app/assets/styles/styles.css:
```diff
@@ -170,3 +170,26 @@ a {
+.donate__btn {
+  font-size: 1.5rem;
+  font-weight: bold;
+  text-decoration: none;
```
I believe this is applied at the global level. Let's remove it for now. If a line appears, I am confident it will go away once we merge everything together :)

misscs commented on index.html:
```diff
@@ -87,7 +87,11 @@ <h1 class="logo-brand" itemprop="brand">MACILE</h1>
   </nav>
-
+  <div class="donate">
+      <h2>Support our growth!</h2>
```
Let's use the custom font for the `h2`. The rest of the text will inherit `Roboto`.

lehysen committed:
Agregando correcciones al index.html y al style.css

misscs commented on app/assets/styles/styles.css:
```diff
@@ -172,24 +172,36 @@ a {
+  margin-left: auto;
+  margin-right: auto;
+}
+.donate__paragraph {
+  width:25em;
+  font-size: 1.5rem;
+  margin-bottom: 1.5em;
+  margin-left: auto;
+  margin-right: auto;
+}
+.donate__h2 {
```
Getting there! 

1. We always have a space between each ruleset in our code base. 
1. We prefer to use semantic class names rather than use the html element. 

There are many examples of how we are naming classes in the code base. Take a look at the bulletin code or some of the PRs #127

misscs replied:
Resolved pairing

misscs commented on app/assets/styles/styles.css:
```diff
@@ -172,24 +172,36 @@ a {
+  margin-right: auto;
+}
+.donate__paragraph {
+  width:25em;
```
Always a space after the semicolon

misscs commented on package.json:
```diff
@@ -24,8 +24,4 @@
-  },
-  "dependencies": {
-    "http-server": "^0.12.3"
-  }
```
You need to run `npm install` to remove the dependency from the `package-lock.json` file. Otherwise it is still technically a dependency of the project.

misscs commented on app/assets/styles/styles.css:
```diff
@@ -172,24 +172,36 @@ a {
   background-color: #6fcf97;
-  padding: 1em 1.5em;
+  padding: 0.5em 1em;
+  display: block;
+  width:3em;
```
If you use `inline-block` you won't need to define a width. Try it out.

misscs reviewed (changes requested):
Getting there! ⭐  Definitely improved, but still work to be done. Take you time. I left many of the same comments on the first review. One suggestion is to write down a list of the changes in the description then you can be sure you capture everything. The github comments are hard to follow sometimes.

misscs committed:
Added a space between rulesets

- Cleaned-up an extra space at the end of a ruleset

misscs committed:
Update class names to be more semantic

misscs committed:
Add space after semicolon

misscs committed:
Remove http-server + fix malformated json

misscs committed:
Move donate css to home.css

- Improve styling
- Use custom font in header
- Remove fixed-width button
- Fix content responsiveness
